chips/rp2040/src/pwm.rs

```diff
@@ -0,0 +1,1290 @@
+    //
+    // Bits 0 to 7 ==> enable channel 0-7 interrupts.
+    fn enable_mask_interrupt(&self, mask: u8) {
+        let old_mask = self.registers.inte.read(CH::CH);
+        self.registers
+            .inte
+            .modify(CH::CH.val(old_mask | mask as u32));
+    }
+
+    // Disable multiple channel interrupts at once.
+    //
+    // Bits 0 to 7 ==> disable channel 0-7 interrupts.
+    fn disable_mask_interrupt(&self, mask: u8) {
+        let old_mask = self.registers.inte.read(CH::CH);
+        self.registers
+            .inte
+            .modify(CH::CH.val(old_mask & !mask as u32));
+    }
+
+    // Clear interrupt flag
+    fn clear_interrupt(&self, channel_number: ChannelNumber) {
+        self.registers
+            .intr
+            .write(CH::CH.val(1 << channel_number as u32));
+    }
+
+    // Force interrupt on the given channel
+    fn force_interrupt(&self, channel_number: ChannelNumber) {
+        let mask = self.registers.intf.read(CH::CH);
+        self.registers
+            .intf
+            .modify(CH::CH.val(mask | 1 << channel_number as u32));
+    }
+
+    // Unforce interrupt
+    fn unforce_interrupt(&self, channel_number: ChannelNumber) {
+        let mask = self.registers.intf.read(CH::CH);
+        self.registers
+            .intf
+            .modify(CH::CH.val(mask & !(1 << channel_number as u32)));
+    }
+
+    // Get interrupt status
+    fn get_interrupt_status(&self, channel_number: ChannelNumber) -> bool {
+        (self.registers.ints.read(CH::CH) & 1 << channel_number as u32) != 0
+    }
+
+    // Handle PWM interrupts
+    //
+    // This method should be called only inside crate::chip::Rp2040DefaultPeripherals::service_interrupt.
+    pub fn handle_interrupt(&self) {
+        for channel_number in CHANNEL_NUMBERS {
+            if self.get_interrupt_status(channel_number) {
+                self.interrupt_handler
+                    .map(|handler| handler.fired(channel_number));
+                self.clear_interrupt(channel_number);
+                self.unforce_interrupt(channel_number);
+            }
+        }
+    }
+
+    // Configure the given channel using the given configuration
+    fn configure_channel(&self, channel_number: ChannelNumber, config: &PwmChannelConfiguration) {
+        self.set_ph_correct(channel_number, config.ph_correct);
+        self.set_invert_polarity(channel_number, config.a_inv, config.b_inv);
+        self.set_div_mode(channel_number, config.divmode);
+        self.set_divider_int_frac(channel_number, config.int, config.frac);
+        self.set_compare_value_a(channel_number, config.cc_a);
+        self.set_compare_value_b(channel_number, config.cc_b);
+        self.set_top(channel_number, config.top);
+        self.set_enabled(channel_number, config.en);
+    }
+
+    /// Set an interrupt handler
+    ///
+    /// See [Interrupt]
+    pub fn set_interrupt_handler(&self, interrupt_handler: &'a dyn Interrupt) {
```

Comment:
```suggestion
    pub(crate) fn set_interrupt_handler(&self, interrupt_handler: &'a dyn Interrupt) {
```

As this is not provided by any HIL, it should either be private either usable only from within the crate.

Reply:
Add a comment stating that this function can only be used from within the crate until we actually have a HIL that exposes this functionality.

Reply:
Removed it since the HIL provides no support for interrupts.